cpp/src/io/orc/writer_impl.hpp

```diff
@@ -286,24 +286,82 @@ class writer::impl {
     hostdevice_2dvector<gpu::encoder_chunk_streams>* enc_streams,
     hostdevice_2dvector<gpu::StripeStream>* strm_desc);
 
-  struct encoded_statistics {
-    std::vector<ColStatsBlob> rowgroup_level;
+  /**
+   * @brief Statistics data stored between calls to write for chunked writes
+   *
+   */
+  struct intermediate_statistics {
+    intermediate_statistics(rmm::cuda_stream_view stream) : stripe_stat_chunks(0, stream){};
+    intermediate_statistics(std::vector<ColStatsBlob> rb,
+                            rmm::device_uvector<statistics_chunk> sc,
+                            hostdevice_vector<statistics_merge_group> smg,
+                            std::vector<statistics_dtype> sdt,
+                            std::vector<data_type> sct)
+      : rowgroup_blobs(std::move(rb)),
+        stripe_stat_chunks(std::move(sc)),
+        stripe_stat_merge(std::move(smg)),
+        stats_dtypes(std::move(sdt)),
+        col_types(std::move(sct)){};
+
+    // blobs for the rowgroups and stripes. Not persisted
+    std::vector<ColStatsBlob> rowgroup_blobs;
+
+    rmm::device_uvector<statistics_chunk> stripe_stat_chunks;
+    hostdevice_vector<statistics_merge_group> stripe_stat_merge;
+    std::vector<statistics_dtype> stats_dtypes;
+    std::vector<data_type> col_types;
+  };
+
+  /**
+   * @brief used for chunked writes to persist data between calls to write.
+   *
+   */
+  struct persisted_statistics {
+    void clear()
+    {
+      stripe_stat_chunks.clear();
+      stripe_stat_merge.clear();
+      stats_dtypes.clear();
+      col_types.clear();
+    }
+
+    std::vector<rmm::device_uvector<statistics_chunk>> stripe_stat_chunks;
+    std::vector<hostdevice_vector<statistics_merge_group>> stripe_stat_merge;
+    std::vector<statistics_dtype> stats_dtypes;
+    std::vector<data_type> col_types;
+    int num_rows;
```

Comment:
just to be on the safe side (otherwise we need a ctor).
```suggestion
    int num_rows = 0;
```